sql/src/main/java/io/crate/executor/transport/TransportShardUpsertAction.java

```diff
@@ -573,16 +556,17 @@ private void processGeneratedColumns(final DocTableInfo tableInfo,
 
     private boolean generatedExpressionEvaluationNeeded(List<Reference> referencedReferences,
                                                         Collection<String> updatedColumns) {
+        boolean evalNeeded = referencedReferences.isEmpty();
```

Comment:
adding a comment here (or as a function javadoc) would be great (like: evaluation is needed either if expression contains no reference at all, or if a referenced column value has changed)